api/types/helmchartargs.go

```diff
@@ -55,10 +55,10 @@ type HelmChart struct {
 	// in the helm template
 	Namespace string `json:"namespace,omitempty" yaml:"namespace,omitempty"`
 
-	// ValuesFile is local file path to a values file to use _instead of_
+	// ValuesFiles are local file paths to a values file to use _instead of_
 	// the default values that accompanied the chart.
 	// The default values are in '{ChartHome}/{Name}/values.yaml'.
-	ValuesFile string `json:"valuesFile,omitempty" yaml:"valuesFile,omitempty"`
+	ValuesFiles []string `json:"valuesFiles,omitempty" yaml:"valuesFiles,omitempty"`
```

Comment:
Unless I'm missing something, this is a breaking API change that ultimately affects the Kustomization schema, which we can't do without a Kustomization version change. So, I think we need to accept both singular and plural inputs and open a frozen issue to remove the singular with Kustomization v1.

Reply:
I think we can refill the singular input value to plural value on [FixKustomization()](https://github.com/kubernetes-sigs/kustomize/blob/ef60d5f9bb3228ff84ad7b44b066ba76de086310/api/types/kustomization.go#L218).
That can delete singular input code and will help deprecate that value in the future.

Reply:
@KnVerey you're right, thanks for the catch. Don't know how I overlooked that.

> I think we can refill the singular input value to plural value on FixKustomization().

Great idea @koba1t! I'll update this PR to make sure we have both the singular and plural values and fix it in FixKustomization().

Reply:
That is a neat idea, but will it work for cases of direct use in the `transformers` field? I feel like to cover that case it might be simpler to have the transformer itself handle the merge.

Reply:
Fixed.